pallets/parachain-staking/src/lib.rs

```diff
@@ -1697,27 +1692,28 @@ pub mod pallet {
 		}
 	}
 
-	/// Add reward points to block authors:
-	/// * 20 points to the block producer for producing a block in the chain
-	impl<T: Config> Pallet<T> {
-		fn award_points_to_block_author() {
-			let author = T::BlockAuthor::get();
+    /// Keep track of number of authored blocks per authority, uncles are counted as well since
+    /// they're a valid proof of being online.
+    impl<T: Config + pallet_authorship::Config>
+      pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
+    {
+        /// Add reward points to block authors:
+        /// * 20 points to the block producer for producing a block in the chain
+        fn note_author(author: T::AccountId) {
```

Comment:
Is this a callback function from the pallet_authorship pallet or are we subscribing to an event?

Reply:
Its neither, we are implementing the `EventHandler` trait from the authorship pallet.